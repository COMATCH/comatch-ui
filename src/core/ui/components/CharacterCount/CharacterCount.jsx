import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    className: PropTypes.string,
    currentCount: PropTypes.number.isRequired,
    label: PropTypes.node,
    maxLength: PropTypes.number.isRequired,
};

const defaultProps = {
    className: '',
    label: 'Remaining characters:',
};

const CharacterCount = ({ className, currentCount, label, maxLength }) => (
    <div className={className}>
        {label} {`${maxLength - currentCount}`}
    </div>
);

CharacterCount.propTypes = propTypes;
CharacterCount.defaultProps = defaultProps;

export { CharacterCount };
