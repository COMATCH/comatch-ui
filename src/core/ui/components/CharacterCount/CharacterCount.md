## CharacterCount

prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**className** | `String` | `''` | :x: | 
**currentCount** | `Number` |  | :white_check_mark: | 
**label** | `ReactNode` | `'Remaining characters:'` | :x: | 
**maxLength** | `Number` |  | :white_check_mark: | 

