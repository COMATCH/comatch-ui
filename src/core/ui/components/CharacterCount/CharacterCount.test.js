import '../../../config/tests/setup';
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { CharacterCount } from './CharacterCount';

describe('CharacterCount', () => {
    it('should render correctly', () => {
        const component = renderer.create(<CharacterCount currentCount={0} maxLength={2000} />).toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should render with default props', () => {
        const component = mount(<CharacterCount currentCount={0} maxLength={2000} />);

        // Required props:
        expect(component.prop('currentCount')).toEqual(0);
        expect(component.prop('maxLength')).toEqual(2000);

        // Optional props:
        expect(component.prop('className')).toEqual('');
        expect(component.prop('label')).toEqual('Remaining characters:');
    });

    it('should render with passed props', () => {
        const component = mount(
            <CharacterCount className="MyClassName" currentCount={0} label="My Label:" maxLength={2000} />,
        );

        expect(component.prop('className')).toEqual('MyClassName');
        expect(component.prop('currentCount')).toEqual(0);
        expect(component.prop('label')).toEqual('My Label:');
        expect(component.prop('maxLength')).toEqual(2000);
    });

    it('should render with correct label/text', () => {
        const component = mount(
            <CharacterCount className="MyClassName" currentCount={1000} label="My Label:" maxLength={2000} />,
        );

        expect(component.text()).toEqual('My Label: 1000');
    });
});
