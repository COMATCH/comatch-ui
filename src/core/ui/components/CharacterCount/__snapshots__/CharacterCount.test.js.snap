// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCount should render correctly 1`] = `
<div
  className=""
>
  Remaining characters:
   
  2000
</div>
`;
